RecoTauTag/RecoTau/plugins/DeepTauId.cc

```diff
@@ -1211,30 +1219,71 @@ class DeepTauId : public deep_tau::DeepTauBase {
       input_layer_ = cache_->getGraph().node(0).name();
       output_layer_ = cache_->getGraph().node(cache_->getGraph().node_size() - 1).name();
       const auto& shape = cache_->getGraph().node(0).attr().at("shape").shape();
-      if (shape.dim(1).size() != dnn_inputs_2017v1::NumberOfInputs)
+      if (shape.dim(1).size() != dnn_inputs_v1::NumberOfInputs)
         throw cms::Exception("DeepTauId")
             << "number of inputs does not match the expected inputs for the given version";
     } else if (version_ == 2) {
-      tauBlockTensor_ = std::make_unique<tensorflow::Tensor>(
-          tensorflow::DT_FLOAT, tensorflow::TensorShape{1, dnn_inputs_2017_v2::TauBlockInputs::NumberOfInputs});
+      using namespace dnn_inputs_v2;
+      namespace sc = deep_tau::Scaling;
+      tauInputs_indices_.resize(TauBlockInputs::NumberOfInputs);
+      std::iota(std::begin(tauInputs_indices_), std::end(tauInputs_indices_), 0);
+
+      if (sub_version_ == 1) {
+        tauBlockTensor_ = std::make_unique<tensorflow::Tensor>(
+            tensorflow::DT_FLOAT, tensorflow::TensorShape{1, TauBlockInputs::NumberOfInputs});
+        scalingParamsMap_ = &sc::scalingParamsMap_v2p1;
+      } else if (sub_version_ == 5) {
+        std::sort(TauBlockInputs::varsToDrop.begin(), TauBlockInputs::varsToDrop.end());
```

Comment:
Why not to write them in order since the beginning at [L215](https://github.com/cms-sw/cmssw/pull/37892/files#diff-c0b5ec61fe6a5ece7dc5cbde414e9572e42b26eaa7baa20a71e9957d4cd4dbc4R215) , instead of having to sort them at evey event?

Reply:
The line in question is in `DeepTauId` [constructor](https://github.com/cms-sw/cmssw/blob/71bf7af2c86c299d87ffd6cf5be806973420cd72/RecoTauTag/RecoTau/plugins/DeepTauId.cc#L1197), so it happens only once per producer instance, not for every event.